processor/schemaprocessor/internal/migrate/conditional.go

```diff
@@ -0,0 +1,92 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migrate // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/migrate"
+
+import "go.opentelemetry.io/collector/pdata/pcommon"
+
+// ValueMatch defines the expected match type
+// that is used on creation of `ConditionalAttributeSet`
+type ValueMatch interface {
+	~string
+}
+
+type ConditionalAttributeSet struct {
+	on    *map[string]struct{}
+	attrs *AttributeChangeSet
+}
+
+type ConditionalAttributeSetSlice []*ConditionalAttributeSet
+
+func NewConditionalAttributeSet[Key AttributeKey, Value AttributeKey, Match ValueMatch](mappings map[Key]Value, matches ...Match) *ConditionalAttributeSet {
+	on := make(map[string]struct{})
+	for _, m := range matches {
+		on[string(m)] = struct{}{}
+	}
+	return &ConditionalAttributeSet{
+		on:    &on,
+		attrs: NewAttributes(mappings),
+	}
+}
+
+func (ca *ConditionalAttributeSet) Apply(attrs pcommon.Map, values ...string) {
```

Comment:
Does `values` need to be a slice? It seems we will always call this with one value.

Reply:
That is something I needed to follow up on, one thing I was sure of is in the span events section, you can supply a set for applies to `span event names` and `span names`. I wasn't sure if they can be combined to restrict to specific values or if they are treated differently.

Reply:
Yes, the set to apply to (the condition) can have multiple elements. That is represented by `ca.on` field. The value to check the conditions against is only a single value (the span name or the event name). That is represented by `values` parameter, right? I don't see a situation when we can have multiple elements in `values`. 
Perhaps I am missing something, I may have misunderstood how the Apply() func is supposed to be used.